.github/workflows/lint-and-test-matrix.yml

```
@@ -51,15 +51,17 @@ jobs:
         python -m pip install --upgrade pip
         pip install -e '.[all]'
 
-    - name: Run Tests with Coverage
+    - name: Run Tests
       run: |
-        pip install coverage pytest pytest-cov
-        coverage run -m pytest --maxfail=1 --disable-warnings -q
-        coverage report -m
-        coverage html
+        pytest --cov --junitxml=junit.xml -o junit_family=legacy --cov-branch --cov-report=xml
```

```suggestion
        pytest --cov
```

Looking at their example app it looks like they do 

```bash
pytest --cov app
```

See: https://github.com/codecov/example-python/blob/main/.github/workflows/ci.yml#L18

I guess in our case it'll be 

```bash
pytest --cov PyTado
```

Although not sure we need to specify the source dir as it's implicit  :thinking:

---

let's test it with this code, this is the current documentation, when I setup codecov. It also includes an experimental testing-overview, ok?

---

Makes sense :+1: